### ISSUE TITLE
Phase A: Implement DataLoader[T] for mini-batch training

### ISSUE THREAD

**Issue by ulises-jeremias** [vlang/vtl#86]
## Problem

VTL has a CIFAR-10 dataset loader but lacks a general-purpose DataLoader that handles batching, shuffling, and iteration.

## Proposed Solution

Implement `DataLoader[T]` struct in `vtl/dataloader.v`:

```v
struct DataLoader[T] {
    dataset   []Tensor[f64]
    batch_size int
    shuffle   bool
    drop_last  bool
    mut indices []int
}

impl[T] DataLoader[T] {
    pub fn next(self) ![]Tensor[f64]
    pub fn len(self) int
    pub fn reset()
}
```

## Acceptance Criteria

- [ ] `DataLoader` struct with configurable `batch_size`, `shuffle`, `drop_last`
- [ ] `next() ![]Tensor[f64]` returns batch of tensors
- [ ] `len() int` returns number of batches
- [ ] `reset()` re-shuffles indices
- [ ] Works with existing `Cifar10Config` output
- [ ] All examples compile with `v -N -W` from `~/.vmodules`

## Dependencies

- Depends on: CIFAR-10 loader (done)
- Blocks: `nn_cifar10` full training loop

## Status

Open — not yet assigned
